File: app/src/main/java/com/jet/mvvmtest/LoginViewModel.java
package com.jet.mvvmtest;

import android.os.Handler;
import android.text.TextUtils;
import android.view.View;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;
import androidx.databinding.library.baseAdapters.BR;
public class LoginViewModel extends BaseObservable {

    @Bindable
    private Boolean clicked = false;

    public Boolean getClicked() {
        return clicked;
    }

    public void setClicked(Boolean clicked) {
        this.clicked = clicked;
        notifyPropertyChanged(BR.clicked);
    }

    public void onLoginClicked() {
        setClicked(true);
    }


}

File: app/src/main/java/com/jet/mvvmtest/LoginViewModel.kt
package com.jet.mvvmtest

import android.os.Handler
import android.text.TextUtils
import android.view.View

import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import androidx.databinding.library.baseAdapters.BR

class LoginViewModel : BaseObservable() {

    @Bindable
    var clicked: Boolean? = false
        set(clicked) {
            field = clicked
            notifyPropertyChanged(BR.clicked)
        }

    fun onLoginClicked() {
        clicked = true
    }


}
